src/api/component.js

```diff
@@ -74,10 +74,11 @@ function callDisconnected(elem) {
 }
 
 // v1
-function Component(self) {
-  const elem = HTMLElement.call(this, self);
-  callConstructor(elem);
-  return elem;
+function Component(...args) {
+  const hasReflect = window.hasOwnProperty('Reflect'); // eslint-disable-line no-prototype-builtins
```

Comment:
This should probably live at the top level as we probably don't need a lazy check for it. You can also do `'Reflect' in window`.